source/adapters/level_zero/v2/command_buffer.cpp

```diff
@@ -0,0 +1,290 @@
+//===--------- command_buffer.cpp - Level Zero Adapter ---------------===//
+//
+// Copyright (C) 2024 Intel Corporation
+//
+// Part of the Unified-Runtime Project, under the Apache License v2.0 with LLVM
+// Exceptions. See LICENSE.TXT
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "command_buffer.hpp"
+#include "../helpers/kernel_helpers.hpp"
+#include "../ur_interface_loader.hpp"
+#include "logger/ur_logger.hpp"
+
+namespace {
+
+// Checks whether zeCommandListImmediateAppendCommandListsExp can be used for a
+// given context.
+void checkImmediateAppendSupport(ur_context_handle_t context) {
+  bool DriverSupportsImmediateAppend =
+      context->getPlatform()->ZeCommandListImmediateAppendExt.Supported;
+
+  if (!DriverSupportsImmediateAppend) {
+    logger::error("Adapter v2 is used but "
+                  "the current driver does not support the "
+                  "zeCommandListImmediateAppendCommandListsExp entrypoint.");
+    std::abort();
+  }
+}
+
+} // namespace
+
+std::pair<ze_event_handle_t *, uint32_t>
+ur_exp_command_buffer_handle_t_::getWaitListView(
```

Comment:
This is identical to the implementation in queue. Please create simple WaitListView abstraction usable in both.